text/0000-lifecycle-experimental-api-mode.md

```diff
@@ -0,0 +1,139 @@
+ - **If** `CNB_BUILDPACK_EXPERIMENTAL_MODE=warn`, **Then** print a warning and continue
+ - **If** `CNB_BUILDPACK_EXPERIMENTAL_MODE=error`, **Then** fail
+ - **If** `CNB_BUILDPACK_EXPERIMENTAL_MODE=silent`, **Then** continue w/o warning
+
+## Lifecycle Descriptor
+
+This RFC proposes the addition of the `experimental` key in `lifecycle.toml` and support for prerelease API versions following the proposed schema:
+
+```
+[apis]
+[apis.buildpack]
+  experimental = ["0.6", "1.0-alpha1"]
+  supported = ["0.5", "0.7-alpha1"]
+  deprecated = ["0.4", "0.5-rc1"]
+[apis.platform]
+  experimental = ["1.0-alpha1"]
+  supported = ["0.6", "0.7-alpha1"]
+  deprecated = ["0.5", "0.6-rc1"]
+
+[lifecycle]
+  version = "0.10.0"
+```
+
+Walking through the Buildpack API side of the example above, the Buildpack API is testing out spec changes for `1.0` with `1.0-alpha1` and `0.7` with `0.7-alpha1`. Buildpack API `0.6` has finalized, but since it included some large features the implementation may under go some changes. Buildpack API `1.0-alpha1` is both experimental in the spec and implementation.
+
+Since experimental versions are defined with a different format, they will NOT automatically get upgraded to a `supported` release when the experimental API has finalized.
+
+### Lifecycle Labels
+This will extend the the `io.buildpacks.lifecycle.apis` label on the builder or lifecycle image with the values contained in the above descriptor file. For example:
+```json
+{
+  "buildpack": {
+    "experimental": ["0.6", "1.0-alpha"],
+    "supported": ["0.5", "0.7-alpha1"],
+    "deprecated": ["0.4", "0.5-rc1"]
+  },
+  "platform": {
+    "experimental": ["1.0-alpha1"],
+    "supported": ["0.6", "0.7-alpha1"],
+    "deprecated": ["0.5", "0.6-rc1"]
+  }
+}
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The technical implementation will follow what's outlined in [RFC#49](https://github.com/buildpacks/rfcs/blob/main/text/0049-multi-api-lifecycle-descriptor.md).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+This will add more complexity/overhead that most users will not use.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Do Nothing
+If we choose to do nothing, we run the risk of releasing less tested API/spec releases.
+
+### Not a Different Version Format
+The Experimental API version format could instead follow the same format as deprecated and supported: `<major>-<minor>`. By doing this, users could automatically get upgraded from an experimental to supported/deprecated API version which may be surprising to them.
+
+### Unstable Features
+Similar to [RFC#91](https://github.com/buildpacks/rfcs/pull/91), instead of tying features to a specific release, features can be tied behind flags for testing. There's a potential this could be complicated to do if the features are coupled to other changes going on in the API/spec. The benefit is folks can try out individual features in a more explicit way.
+
+### Experimental Section in the API
```

Comment:
From the WG today, @sclevine requested this be moved into the actual proposal.

Reply:
If we pull in unstable features do we need something like `CNB_PLATFORM_UNSTABLE_MODE` with `error`, `warn` and `silent` options?

Reply:
Assuming we go ahead with the unstable features, I think it is potentially confusing that "experimental" in `pack` is analogous to "unstable" in lifecycle, while "experimental" in lifecycle is an entirely different concept.

I would prefer to call the unstable features experimental features and leave the experimental API as documented in this RFC out for now. Having so many different mechanism for experimental support seems like it could be potentially confusing to end users imho.

Reply:
@ekcasey I moved experimental features up into the proposal and moved experimental "mode" out into alternatives. Do you still have naming concerns?

Reply:
looks good!